### PR TITLE
Action Manager | Edit Menu Migration

### PR DESCRIPTION
**Pull request by AMZN-daimini**
Migrates the Edit menu in the Editor main window to the new Action Manager.

image

Some extra work was needed to hook up actions to the correct notifications to update their enabled and checked state, since the legacy system did not have them and would just poll for changes.

### CODE REVIEW

**Comment by AMZN-daimini** [comment]
> > @hultonha Hey there, I am working on porting over the actions for the Edit menu and I need some sort of notification for whenever some of the settings that are controlled by those menu items are changed. You can check them all out searching for TODOs, but here's the ones that I'm not sure how to handle right now:
> > ```
> > // TODO - trigger update when the angle snapping changes?
> > // TODO - trigger update when the grid snapping changes?
> > // TODO - Update when the transform mode changes.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Do we have a notification bus for these properties so that the actions can be updated even if those values are changed somewhere else? (Note that the legacy Action Manager did not need this since actions were updated 4 times per frame via polling, and I'm trying to move away from that). The alternative is to make those menu items the only way to change those properties, or updated them via polling again (but I'd like not to :D )
> > Thanks!
> 
> Hey @AMZN-daimini! Great question, I'm trying to remember all the details specifically (it's been a little while 😅) but I think you _can_ listen for a notification. I believe we're using an AZ::Event to do this in `EditorViewportSettings.cpp` (see `EditorViewportSettingsCallbacksImpl`). That definitely has a `SetGridSnappingChangedEvent` for this purpose (not 100% sure about angle snapping). We might be able to catch-up briefly today about this if you're around (I'll ping you when I'm free) otherwise it might be easiest to chat next week (or the one after due to us all being busy next week). I hope the info I provided helps a little in the interim. Thanks!

Nice, thanks for the leads. I will give this a look :D Do hit me up if you have some time tho so we can double check :)

File: Code/Editor/Core/EditorActionsHandler.cpp [hunk 1]
```diff
@@ -435,6 +449,154 @@ void EditorActionsHandler::InitializeActions()
...
+    {
+        AzToolsFramework::ActionProperties actionProperties;
+        actionProperties.m_name = "&Undo";
+        actionProperties.m_description = "Undo last operation";
+        actionProperties.m_category = "Edit";
+        actionProperties.m_hideFromMenusWhenDisabled = false;
+
+        m_actionManagerInterface->RegisterAction(
+            EditorMainWindowActionContextIdentifier,
+            "o3de.action.edit.undo",
```
Comment (jonawals) [inline Code/Editor/Core/EditorActionsHandler.cpp:464]: Q: `"o3de.action.edit.undo"` can this not be a named constant like the others above? Same for the others below.

**Review by jonawals — APPROVED** [review]
LGTM

File: Code/Editor/Core/EditorActionsHandler.cpp (hunk 1 above)
Comment (AMZN-daimini) [reply Code/Editor/Core/EditorActionsHandler.cpp:464]: It can, but there's so many :'D

**Review by alexmontAmazon — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

**Pull request merged**